system/classes/KO7/REST/Format/XML.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class for formatting REST-Response bodies as XML
+ *
+ * @package        KO7\REST
+ *
+ * @copyright  (c) since 2016 Koseven Team
+ * @license        https://koseven.ga/LICENSE
+ */
+class KO7_REST_Format_XML extends REST_Format {
+
+    /**
+     * Format function
+     *
+     * @throws REST_Exception
+     *
+     * @return string
+     */
+    public function format() : string
+    {
+        // Check if php-xml is loaded
+        if ( ! extension_loaded('xml'))
+        {
+            throw new REST_Exception('PHP XML Module not loaded.');
+        }
+
+        // Create new XML Element
+        $xml = $this->array_to_xml($this->_body);
+
+        // Check if xml is valid
+        if ( ! $xml)
+        {
+            throw new REST_Exception($this->evaluate_error());
+        }
+
+        return $xml;
+    }
+
+    /**
+     * Convert Array to an xml element
+     *
+     * @param array $array        Array to convert
+     * @param mixed $rootElement  Root element of entry
+     * @param null $xml           Current XML stack
+     *
+     * @return mixed
+     */
+    protected function array_to_xml(array $array, $rootElement = NULL, $xml = NULL)
+    {
+        // Use internal error handling
+        libxml_use_internal_errors(true);
```

Comment:
Pls, if u dont understand then ask me
As I say, need add `LIBXML_NOERROR | LIBXML_NOWARNING`. You can ask me - why not libxml_use_internal_errors? Because u CHANGE GLOBAL SETTING

Reply:
I've read about some issues using `LIBXML_NOERROR`, very common in php therefore i used the function, i understand the issue of changing a global setting, that's why i am backing up the old setting and restoring it afterwards now.